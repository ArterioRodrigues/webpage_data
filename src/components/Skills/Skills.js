import React from 'react';
import Container from '../Container/Container';
import skills from '../../data/skills.json';

const Skills = () => (
  <Container>
    <div className="skills">
      <h2 className="skills__title">skills</h2>
      {skills.map(section => (
        <div className="skills__subsection" key={section.title}>
          <h3 className="skills__subtitle">{section.title}</h3>
          <div className="skills__items">
            <ul>
              {section.items.map(item => (
                <li key={item}>
                  <span>{item}</span>
                </li>
              ))}
            </ul>
          </div>
        </div>
      ))}
    </div>
  </Container>
);

export default Skills;
