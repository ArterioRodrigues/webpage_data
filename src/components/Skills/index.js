/**
 * Skills component.
 */
import React from 'react';
import Container from '../Container';
import skills from '../../data/skills.json';

const Skills = () => (
  <Container title={skills.sectionTitle}>
    {skills.content.map((subSection) => (
      <div key={subSection.title}>
        <h3>{subSection.title}</h3>
        <ul>
          {subSection.items.map((item) => (
            <li key={item}>
              <span>{item}</span>
            </li>
          ))}
        </ul>
      </div>
    ))}
  </Container>
);

export default Skills;
