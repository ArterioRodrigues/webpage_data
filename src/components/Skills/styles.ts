/**
 * Skills component styles.
 */
import { Styles } from 'react-jss';
import containerStyles from '../Container/styles';

export default {
  subSectionTitle: containerStyles.subtitle,
} as Styles;
