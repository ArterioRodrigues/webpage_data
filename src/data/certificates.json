{
  "sectionTitle": "certifications",
  "sectionSubtitle": "Certificates that I have collected over time",
  "content": [
    {
      "name": "Responsive Web Design Certification",
      "url": "https://www.freecodecamp.org/certification/caglarturali/responsive-web-design"
    },
    {
      "name": "Introduction to Game Development",
      "url": "https://www.coursera.org/account/accomplishments/certificate/GKDBNAGKPD6M"
    },
    {
      "name": "Introduction to Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/UC3L666SD5TG"
    },
    {
      "name": "3D Models for Virtual Reality",
      "url": "https://www.coursera.org/account/accomplishments/certificate/4YHK3LYWMJPV"
    },
    {
      "name": "The Art of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/DNVYLQQL3WVF"
    },
    {
      "name": "The Technology of Music Production",
      "url": "https://www.coursera.org/account/accomplishments/certificate/VZDUNEC24XER"
    },
    {
      "name": "Pro Tools Basics",
      "url": "https://www.coursera.org/account/accomplishments/certificate/YW4QY26UHC5F"
    }
  ]
}
