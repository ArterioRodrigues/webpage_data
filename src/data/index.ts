export * from './certificates';
export * from './contact';
export * from './education';
export * from './greeter';
export * from './metadata';
export * from './projects';
export * from './skills';
