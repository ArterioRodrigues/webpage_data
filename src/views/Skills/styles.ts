/**
 * Skills component styles.
 */
import { Styles } from 'react-jss';
import containerStyles from '../../components/Container/styles';

export default {
  subSectionTitle: containerStyles.subtitle,
} as Styles;
