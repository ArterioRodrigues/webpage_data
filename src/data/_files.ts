import { NavLink } from '../models';
import Greeter from '../views/Greeter';
import Skills from '../views/Skills';
import Education from '../views/Education';
import Projects from '../views/Projects';
import Certificates from '../views/Certificates';
import Contact from '../views/Contact';
import { certificates, contact, education, projects, skills } from '.';

export const files: NavLink[] = [
  {
    icon: ['fab', 'html5'],
    color: '#e44d26',
    url: '/',
    name: 'index.html',
    route: {
      component: Greeter,
      data: contact,
    },
  },
  {
    icon: ['fab', 'js'],
    color: '#ffca28',
    url: '/skills',
    name: 'skills.js',
    route: {
      component: Skills,
      data: skills,
    },
  },
  {
    icon: 'university',
    color: '#42a5f5',
    url: '/education',
    name: '.educationrc',
    route: {
      component: Education,
      data: education,
    },
  },
  {
    icon: 'sliders-h',
    color: '#98c379',
    url: '/projects',
    name: 'projects.config',
    route: {
      component: Projects,
      data: projects,
    },
  },
  {
    icon: ['fab', 'sass'],
    color: '#e13e76',
    url: '/certifications',
    name: 'certifications.sass',
    route: {
      component: Certificates,
      data: certificates,
    },
  },
  {
    icon: ['fab', 'markdown'],
    color: '#abb2bf',
    url: '/contact',
    name: 'contact.md',
    route: {
      component: Contact,
      data: contact,
    },
  },
];
