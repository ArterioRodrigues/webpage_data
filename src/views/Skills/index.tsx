/**
 * Skills component.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import Container from '../../components/Container';
import { Page, Skill } from '../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export interface SkillsProps {
  data: Page<Skill>;
}

const Skills: React.FC<SkillsProps> = ({ data }) => {
  const classes = useStyles();

  return (
    <Container title={data.name}>
      {data.content.map(({ title, items }) => (
        <div key={title}>
          <h3 className={classes.subSectionTitle}>{title}</h3>
          <ul>
            {items.map((item) => (
              <li key={item}>
                <span>{item}</span>
              </li>
            ))}
          </ul>
        </div>
      ))}
    </Container>
  );
};

export default Skills;
