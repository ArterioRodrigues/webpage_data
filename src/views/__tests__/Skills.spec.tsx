import React from 'react';
import { render } from '@testing-library/react';
import Skills from '../Skills';
import { skills } from '../../data';

describe('<Skills />', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <Skills data={skills} fileName="skills.test" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
