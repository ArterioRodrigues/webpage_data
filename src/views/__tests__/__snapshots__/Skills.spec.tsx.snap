// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root-0-2-2"
  >
    <h2
      class="title-0-2-3"
    >
      skills
    </h2>
    <h3
      class="subtitle-0-2-4"
    />
    <div
      class="content-0-2-5"
    >
      <div>
        <h3
          class="subSectionTitle-0-2-1"
        >
          Technical skills
        </h3>
        <ul>
          <li>
            <span>
              Web Development (End-to-End)
            </span>
          </li>
          <li>
            <span>
              Responsive Design
            </span>
          </li>
          <li>
            <span>
              Mobile Development (Hybrid/Native)
            </span>
          </li>
          <li>
            <span>
              Game Development (Unity 3D)
            </span>
          </li>
          <li>
            <span>
              VR Development (Unity 3D)
            </span>
          </li>
          <li>
            <span>
              3D Modelling / Animation
            </span>
          </li>
        </ul>
      </div>
      <div>
        <h3
          class="subSectionTitle-0-2-1"
        >
          Programming languages / tech stacks
        </h3>
        <ul>
          <li>
            <span>
              JavaScript/TypeScript
            </span>
          </li>
          <li>
            <span>
              React/React Native
            </span>
          </li>
          <li>
            <span>
              Node.js
            </span>
          </li>
          <li>
            <span>
              Express.js
            </span>
          </li>
          <li>
            <span>
              HTML & CSS
            </span>
          </li>
          <li>
            <span>
              MongoDB
            </span>
          </li>
          <li>
            <span>
              MySQL
            </span>
          </li>
          <li>
            <span>
              Docker
            </span>
          </li>
          <li>
            <span>
              Golang (newcomer)
            </span>
          </li>
          <li>
            <span>
              Java (~1 year)
            </span>
          </li>
          <li>
            <span>
              C#/.NET (3+ years)
            </span>
          </li>
          <li>
            <span>
              PHP (~3 years)
            </span>
          </li>
        </ul>
      </div>
      <div>
        <h3
          class="subSectionTitle-0-2-1"
        >
          Preferences
        </h3>
        <ul>
          <li>
            <span>
              elementary OS üêß
            </span>
          </li>
          <li>
            <span>
              Zshell üíª
            </span>
          </li>
          <li>
            <span>
              VSCode ¬Ø\\_(„ÉÑ)_/¬Ø
            </span>
          </li>
        </ul>
      </div>
      <div>
        <h3
          class="subSectionTitle-0-2-1"
        >
          Special interests
        </h3>
        <ul>
          <li>
            <span>
              Electronic music production
            </span>
          </li>
          <li>
            <span>
              Entrepreneurship
            </span>
          </li>
          <li>
            <span>
              Technical documentaries
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
